### ISSUE TITLE
Compilation fails lead to kernel restarting

### ISSUE THREAD

**Issue by ioanaif** [compiler-research/xeus-clang-repl#22]
<img width="921" alt="image" src="https://user-images.githubusercontent.com/9751871/220171954-07fd72b9-bb45-44ec-8f33-372dcf641fd0.png">

In the example above, the kernel simply restarts. Better recovery from exceptions / errors should be implemented so that the kernel doesn't automatically restart and lose its state.

**Comment by p4vook** [comment]
This is most definitely an upstream (clang-repl) bug

**Comment by vgvassilev** [comment]
@ioanaif, can you still reproduce this?

**Comment by ioanaif** [comment]
Yes. However, the idea here was not to flag an error; But the fact that the whole kernel restarts which means that the user loses the state of the kernel and needs to rerun all cells. 

The idea behind this issue was to state that it might be a good policy to implement a wrapper which catches different types of compiler errors and prevents the kernel from restarting and instead it could output an informative error message.
